addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/test/YamahaEmulatedHttpXMLServer.java

```diff
@@ -0,0 +1,286 @@
+package org.openhab.binding.yamahareceiver.test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.openhab.binding.yamahareceiver.internal.protocol.HttpXMLSendReceive;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.xml.sax.InputSource;
+
+/**
+ * Emulates a Yamaha Receiver API endpoint (Http/XML on localhost:12121/YamahaRemoteControl/ctrl)
+ *
+ */
+public class YamahaEmulatedHttpXMLServer implements Runnable {
```

Comment:
@davidgraeff You can simply move all test classes into a separate source folder (src/test/java, which outputs to target/classes) within the same project. This will make sure that the packaged bundle won't include the test code (and I second @MHerbst that that code shouldn't be shipped with the binary).